### PR TITLE
Fixing OpenCL download

### PR DESCRIPTION
**Pull request by henryiii**
Fixing the OpenCL build on Travis. The URL changed, breaking the download script. This also adds `-j2` to the build procedure and removes sudo, which should speed up the build a bit.

### CODE REVIEW

**Comment by henryiii** [comment]
Any idea how to fix the final error (clang JIT missing libomp.so)? I thought it was a CMake issue but now think it's a path issue. I can grab a Docker container and check it in a day or two otherwise.

**Comment by henryiii** [comment]
This might be a bug in the Find OpenMP module in CMake. I think it's grabbing the first option it tries (`-fopenmp=libomp`), instead of the second attempt (`-fopenmp=libiomp5`), which is correct for 14.04 (not sure if it needs to be installed, though). I'll check this at some point and see if that's what is happening.

**Comment by ddemidov** [comment]
It looks like `libomp.so` on Ubuntu is installed into non-standard location, so exporting the correct `LD_LIBRARY_PATH` seems to be enough (see c6b55bc).

File: .travis.yml [hunk 1]
```diff
@@ -30,22 +28,27 @@ addons:
...
-
+      - python-pip
 cache:
     directories:
         - ${HOME}/${BOOST_BASENAME}/boost
         - ${HOME}/${BOOST_BASENAME}/stage/lib
         - ${AMDAPPSDKROOT}
 
 before_install:
+    - if [ "$VEXCL_BACKEND" = "JIT" ] && [ "$CC" = "clang" ] ; then pip install --user cmake ; fi
```
Comment (ddemidov) [inline .travis.yml]: Do we still need to install the latest cmake (and pip in line 31)? Otherwise, this looks ready to merge.

Comment (henryiii) [reply .travis.yml]: No, but it might be nice to have a test with the latest CMake too. No reason for it to be this one, though, so I'll try removing this.

File: CMakeLists.txt [hunk 2]
```diff
@@ -137,15 +137,24 @@ if (NOT "${Boost_VERSION}" STRLESS "106100")
...
+            message(WARNING "Clang JIT needs CMake 3.9 to properly detect OpenMP linker flags!")
+        endif()
+        # Clang may need -fopenmp=libiomp5 instead, can't be detected here without CMake 3.9
+        target_link_libraries(JIT INTERFACE
+            $<$<CXX_COMPILER_ID:GNU>:${OpenMP_CXX_FLAGS}>
+            $<$<CXX_COMPILER_ID:Clang>:-fopenmp=libomp>
+            $<$<CXX_COMPILER_ID:Intel>:${OpenMP_CXX_FLAGS}>
+            )
+        target_compile_options(JIT INTERFACE ${OpenMP_CXX_FLAGS})
+    endif()
```
Comment (ddemidov) [inline CMakeLists.txt]: Also, is this change necessary now?

Comment (henryiii) [reply CMakeLists.txt]: Yes, it is better to use the `OpenMP::OpenMP_CXX` target if it exists. I've scaled back some of the changes, though, to keep this clean.

Comment (henryiii) [reply CMakeLists.txt]: On some systems, like Windows as you've seen, linker and compiler flags are different. CMake knows that and handles it appropriately through FindOpenMP (for CMake 3.9+), instead of having to list each one like is done now.

**Comment by henryiii** [comment]
Looks good to me too. Please merge (or squash and merge?) and then I'll rebase the other PR on this.

**Comment by ddemidov** [comment]
Thank you!
